src/Upgrader/Business/VersionControlSystem/GitVcs.php

```diff
@@ -48,82 +53,162 @@ public function __construct(UpgraderConfig $config, ProviderInterface $provider)
+     * @return \Upgrader\Business\VersionControlSystem\Response\Collection\VcsResponseCollection
+     */
+    public function rollback(): VcsResponseCollection
+    {
+        $collection = new VcsResponseCollection();
+        $collection->add($this->deleteLocalBranch($this->getHeadBranch()));
+        $collection->add($this->deleteRemoteBranch($this->getHeadBranch()));
+        if ($this->isUncommittedChangesExist()) {
+            $collection->addCollection($this->revertUncommittedChanges());
+        }
+
+        return $collection;
+    }
+
+    /**
+     * @return \Upgrader\Business\VersionControlSystem\Response\VcsResponse
+     */
+    public function checkout(): VcsResponse
+    {
+        $command = ['git', 'checkout', $this->getBaseBranch()];
+        $process = $this->runProcess($command);
+
+        return $this->createResponseForProcess($process);
     }
 
     /**
      * @param string $message
      *
      * @return \Upgrader\Business\VersionControlSystem\Response\VcsResponse
      */
-    public function commit(string $message): VcsResponse
+    public function commitChanges(string $message): VcsResponse
     {
         $command = ['git', 'commit', '-m', $message];
         $process = $this->runProcess($command);
 
-        return $this->createResponse($process);
+        return $this->createResponseForProcess($process);
     }
 
     /**
      * @param string $branch
      *
      * @return \Upgrader\Business\VersionControlSystem\Response\VcsResponse
      */
-    public function push(string $branch): VcsResponse
+    public function pushChanges(string $branch): VcsResponse
     {
-        $remote = sprintf(
-            'https://%s@github.com/%s/%s.git',
-            $this->config->getGithubAccessToken(),
-            $this->config->getGithubOrganization(),
-            $this->config->getGithubRepository()
-        );
-        $command = ['git', 'push', '--set-upstream', $remote, $branch];
+        $command = ['git', 'push', '--set-upstream', $this->getRemote(), $branch];
         $process = $this->runProcess($command);
-        $command[3] = substr_replace($remote, str_repeat('*', 10), 8, strpos($remote, '@') - 8);
+        $command[3] = $this->getPublicRemote();
 
-        return $this->createResponse($process, implode(' ', $command));
+        return $this->createResponseForProcess($process, implode(' ', $command));
     }
 
     /**
      * @return \Upgrader\Business\VersionControlSystem\Response\VcsResponse
      */
-    public function check(): VcsResponse
+    public function checkUncommittedChanges(): VcsResponse
     {
-        $command = ['git', 'update-index', '--refresh'];
+        if ($this->isUncommittedChangesExist()) {
+            return $this->createResponse(false, 'You have to fix uncommitted changes');
+        }
+
+        return $this->createResponse(true, "You don't have uncommitted changes");
+    }
+
+    /**
+     * @return bool
+     */
+    public function isUncommittedChangesExist(): bool
```

Comment:
is...Exist is an invalid naming as per our conventions, and double so with plural in there

hasUncommitedChanges